src/libstore/build/local-derivation-goal.cc

```diff
@@ -415,6 +416,7 @@ void LocalDerivationGoal::startBuilder()
             throw Error("cannot determine cgroup name from /proc/self/cgroup");
 
         auto ourCgroupPath = canonPath("/sys/fs/cgroup/" + ourCgroup);
+        ourCgroupPath = dirOf(ourCgroupPath);
```

Comment:
This is very confusing naming. Doing `dirOf` means it *isn't* our cgroup, but the parent's. And it's not clear that we want to be putting stuff in the parent cgroup, since it causes nix to break out of whatever resource controls are in place for its own cgroup. It especially wouldn't be nice if Nix builds escape from `system.slice/nix-daemon.service`.

Reply:
i had modified things so the daemon runs in `system.slice/nix-daemon.service/actual-daemon`

the problem is that you cant certain things on `system.slice/nix-daemon.service` if it contains any process, everything must be in a child c-group

i'm not sure what the best solution to that is, and this currently needs an extra script around `nix-daemon` to move it to a child cgroup